docs2/site/docs/migrations/migration4.md

```diff
@@ -322,3 +322,4 @@ lock (field)
 * `ExecutionNode.Source` is read-only; additional derived classes have been added for subscriptions
 * `NameValidator.ValidateName` and `NameValidator.ValidateNameOnSchemaInitialize` accept an enum instead of a string for their second argument
 * `ExecutionNode.PropagateNull` must be called before `ExecutionNode.ToValue`; see reference implementation
+* `IResolveFieldContext.FieldName` and `IResolveFieldContext.ReturnType` properties have been removed, use `IResolveFieldContext.FieldAst.Name` and `IResolveFieldContext.FieldDefinition.ResolvedType` instead
```

Comment:
This has nothing to do with this PR.  I do not think they should be removed, but if we want to discuss it, we should do it separately from this PR.

Further, there is no comment that `Parent` has been added in the 'new features' section.

Reply:
Two removed properties are always calculated and do not carry their own semantic meaning. They should be removed.
I definitely added a note about `Parent` in some other PR.

Reply:
Yes, in #2276 . I don't want to aggravate the situation with merge conflicts.

Reply:
>  I do not think they should be removed

See here:
![изображение](https://user-images.githubusercontent.com/21261007/108522923-62c02d80-72de-11eb-8f04-3c4328d31dcb.png)

These properties should be removed to avoid the problem of invalid data. They are not just overkill, they are the cause of potential error. Now you can set values that contradict each other.

Reply:
I added note about `Parent`.

Reply:
The properties are convenience properties on `IResolveFieldContext`, just like almost all of the properties are.  We both know that `ReadonlyResolveFieldContext` only holds a few members.  But many properties are provided (mapped) so the user can quickly access the fields they care about.  Further, `IResolveFieldContext` is one of the most used classes in the project.  We're not talking about some obscure part of the guts of the project.  We're talking about code that everyone deals with in every graph in every project.  If we want to eliminate these properties, let's mark them as obsolete for v4 and remove them in v5.  It does not benefit anything to remove them now.  It is not faster, it does not save memory, it does not help reorganize the API for the better, nothing.  I agree they are redundant, but they are not out of place.  Removing them only servers one purpose: adding to user frustration when they convert to v4.

Reply:
I have no problem if you want to change `ResolveFieldContext.FieldName` to be a mapping instead of a read/write property.  That class is almost exclusively used for tests.

Reply:
> it does not help reorganize the API for the better

It does. We declared interface with two superfluous properties. This interface smells.

I don't see a problem in removing them, we made a lot of changes and we have a dood migration guide. Why drag into the new version something that is absolutely unnecessary now? Customers will still need to recompile their code.

Reply:
Ok

Reply:
Let's at least mark them obsolete in 3.x then, with a note saying the properties to use instead.

Reply:
#2301